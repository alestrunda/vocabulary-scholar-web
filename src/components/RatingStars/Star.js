import React from 'react';
import PropTypes from 'prop-types';
import ToggleStar from 'material-ui/svg-icons/toggle/star';

const RatingStar = props => {
  const handleClick = () => {
    props.onClick && props.onClick(props.number);
  };

  const handleMouseEnter = () => {
    props.onMouseEnter && props.onMouseEnter(props.number);
  };

  const handleMouseLeave = () => {
    props.onMouseLeave && props.onMouseLeave(props.number);
  };

  return (
    <div
      onClick={handleClick}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
      className={`rating-stars__star rating-stars__star--${
        props.active ? 'active' : 'unactive'
      }`}
      style={props.style}
    >
      <ToggleStar
        className="rating-stars__icon"
        color={props.color}
        style={{
          width: props.size,
          height: props.size,
          ...props.styleStar,
        }}
      />
    </div>
  );
};

RatingStar.propTypes = {
  number: PropTypes.number,
  size: PropTypes.number,
  color: PropTypes.string,
  onClick: PropTypes.func,
  onMouseEnter: PropTypes.func,
  onMouseLeave: PropTypes.func,
  style: PropTypes.object,
  styleStar: PropTypes.object,
  active: PropTypes.bool,
};

RatingStar.defaultProps = {
  active: false,
};

export default RatingStar;
