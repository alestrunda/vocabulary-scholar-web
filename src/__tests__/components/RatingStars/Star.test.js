import React from 'react';
import { shallow } from 'enzyme';

import Star from '../../../components/RatingStars/Star';

it('rendes correctly', () => {
  const star = shallow(<Star />);
  expect(star).toMatchSnapshot();
});

it('renders star', () => {
  const size = 20,
    color = '#f00';
  const star = shallow(<Star size={size} color={color} />);
  expect(star.find('ToggleStar').prop('style')).toHaveProperty('width', size);
  expect(star.find('ToggleStar').prop('color')).toBe(color);
});

it('triggers onClick', () => {
  const starNumber = 3,
    onClick = jest.fn();
  const star = shallow(<Star onClick={onClick} number={starNumber} />);
  star.simulate('click');
  expect(onClick).toHaveBeenCalledWith(starNumber);
});

it('triggers onMouseEnter', () => {
  const starNumber = 4,
    onMouseEnter = jest.fn();
  const star = shallow(
    <Star onMouseEnter={onMouseEnter} number={starNumber} />
  );
  star.simulate('mouseenter');
  expect(onMouseEnter).toHaveBeenCalledWith(starNumber);
});

it('triggers onMouseLeave', () => {
  const starNumber = 2,
    onMouseLeave = jest.fn();
  const star = shallow(
    <Star onMouseLeave={onMouseLeave} number={starNumber} />
  );
  star.simulate('mouseleave');
  expect(onMouseLeave).toHaveBeenCalledWith(starNumber);
});
