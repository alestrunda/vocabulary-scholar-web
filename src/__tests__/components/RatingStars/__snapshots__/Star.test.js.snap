// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendes correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RatingStar
    active={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ToggleStar
        className="rating-stars__icon"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />,
      "className": "rating-stars__star rating-stars__star--unactive",
      "onClick": [Function],
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
      "style": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "rating-stars__icon",
        "color": undefined,
        "style": Object {
          "height": undefined,
          "width": undefined,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ToggleStar
          className="rating-stars__icon"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />,
        "className": "rating-stars__star rating-stars__star--unactive",
        "onClick": [Function],
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "rating-stars__icon",
          "color": undefined,
          "style": Object {
            "height": undefined,
            "width": undefined,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
